pubsub/spring/src/main/java/demo/PubSubApplication.java

```diff
@@ -91,18 +90,8 @@ public void messageReceiver(
   public MessageHandler messageSender(PubSubTemplate pubsubTemplate) {
     PubSubMessageHandler adapter = new PubSubMessageHandler(pubsubTemplate, "topic-two");
 
-    adapter.setPublishCallback(
-        new ListenableFutureCallback<String>() {
-          @Override
-          public void onFailure(Throwable throwable) {
-            LOGGER.info("There was an error sending the message.");
-          }
-
-          @Override
-          public void onSuccess(String result) {
-            LOGGER.info("Message was sent via the outbound channel adapter to topic-two!");
-          }
-        });
+    adapter.setFailureCallback(
```

Comment:
Do you want to also add a success callback to match the behavior of the old code?

Reply:
Oh yes. Thanks for noticing that. Done.